bundles/org.openhab.automation.jsscripting/src/main/java/org/openhab/automation/jsscripting/internal/OpenhabGraalJSScriptEngine.java

```diff
@@ -337,7 +357,30 @@ protected String onScript(String script) {
             logger.debug("Injecting event conversion code into script for engine '{}'.", engineIdentifier);
             newScript = EVENT_CONVERSION_CODE + System.lineSeparator() + newScript;
         }
-        if (configuration.isWrapperEnabled()) {
+
+        // keep this extendable for more directives by checking the first n lines (n = number of directives)
+        // up to two directives: "use strict" (handled by Graal) and "use wrapper"
+        List<String> header = script.lines().limit(2).toList();
+        boolean useWrapper = configuration.isScriptConditionWrapperEnabled();
+        for (String line : header) {
+            var matcher = USE_WRAPPER_DIRECTIVE.matcher(line);
+            if (!matcher.matches()) {
+                continue;
+            }
+            var enabled = matcher.group("enabled");
+            if (enabled == null || enabled.isBlank()) {
+                useWrapper = true;
+            } else if ("false".equals(enabled)) {
+                useWrapper = false;
+            } else if ("true".equals(enabled)) {
+                useWrapper = true;
+            } else {
+                logger.warn("Invalid value '{}' for 'use wrapper' directive in script for engine '{}'.", enabled,
+                        engineIdentifier);
+            }
+        }
+
+        if (isScriptAction() || (isScriptCondition() && useWrapper)) {
```

Comment:
I think you forgot to disable the wrapper for actions if the directive disables it.

Reply:
Correct, thanks for the reminder. 
Now I recall we discussed that.

Reply:
Just pushed the change :+1: